app/build.gradle

```diff
@@ -103,13 +103,13 @@ task jacocoTestReport(type: JacocoReport, dependsOn: ['testDebugUnitTest', 'crea
         html.enabled = true
     }
 
-    def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test*.*', 'android/**/*.*']
-    def debugTree = fileTree(dir: "$project.buildDir/intermediates/classes/debug", excludes: fileFilter)
+    def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test*.*', 'test/**/*.*', '**/*test*.*', 'android/**/*.*']
+    def debugTree = fileTree(dir: "$project.buildDir/intermediates/javac/debug/classes", excludes: fileFilter)
     def mainSrc = "$project.projectDir/src/main/java"
 
     sourceDirectories.from = [mainSrc]
     classDirectories.from = [debugTree]
     executionData.setFrom(project.fileTree(dir: project.buildDir, includes: [
-            'jacoco/testDebugUnitTest.exec', 'outputs/code-coverage/connected/*coverage.ec'
+            'jacoco/testDebugUnitTest.exec', 'outputs/code-coverage/connected/*coverage.ec', 'outputs/code_coverage/debugAndroidTest/connected/*coverage.ec'
```

Comment:
You can remove the middle one, since this file doesn't exist it could mess things up.

Reply:
I also see that in dependencies you have: `testImplementation "org.junit.jupiter:junit-jupiter-api:5.3.2"`
Try changing that to `testImplementation 'junit:junit:4.12'`

Reply:
We wanted to take advantage of some features from jupited/junit5 (that don't exist in junit4)